### PR TITLE
Change API to Addon instead of mixin

### PR DESCRIPTION
**Pull request by rxaviers**
Depends on:
- #8

TODO:
- [ ] Update docs accordingly.

### CODE REVIEW

**Comment by rxaviers** [comment]
This is specially interesting: https://github.com/kborchers/react-globalize/pull/7/files#diff-406807581e0b49f296c2beb81859ba21R51

**Comment by rxaviers** [comment]
Examples updated to reflect addons, AMD one runnable. CJS one almost runnable.

File: examples/amd/bower.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "react-globalize-example",
+  "devDependencies": {
+    "almond": "~0.3.1",
+    "cldr-data": "27.x",
+    "globalize": "jquery/globalize#fix-398-runtime",
```
Comment (rxaviers) [inline examples/amd/bower.json]: Temporary

File: examples/amd/bower.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    "react": "0.13.1",
+    "react-globalize": "rxaviers/react-globalize#addons",
```
Comment (rxaviers) [inline examples/amd/bower.json]: Temporary

File: examples/cjs/package.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    "browserify": "^6.3.3",
+    "node-jsx": "^0.12.4",
+    "reactify": "^0.17.1",
+    "uglify-js": "^2.4.20"
+  },
+  "dependencies": {
+    "cldr-data": "27.x",
+    "react": "~0.13.0",
+    "react-globalize": "rxaviers/react-globalize#addons",
+    "globalize": "jquery/globalize#fix-398-runtime"
```
Comment (rxaviers) [inline examples/cjs/package.json]: Temporary

**Comment by kborchers** [comment]
So I feel like this is a bit much for a single PR. Can you break out b8bbb14ce5f7eee0e8b22188d3da02fc291aae27 into its own PR as well as the AMD/CJS stuff into another PR so that there would be a total of 3 PRs? I would like to review each of those separately.

**Comment by kborchers** [comment]
OK, I merged the other PRs so if you could break this into 1 PR for the Mixin->Addon change and another PR for the AMD/CJS stuff and rebase both that would be great.

**Comment by rxaviers** [comment]
Absolutely, this needs to be broken down. Doing it...

**Comment by rxaviers** [comment]
Rebased, only the addons change in here now. It still needs docs update.

**Comment by kborchers** [comment]
In general this looks good. I wonder if there is a more elegant way to do this https://github.com/rxaviers/react-globalize/commit/fae8f30346a9682849228ce13c78a94c1857c940#diff-168726dbe96b3ce427e7fedce31bb0bcR23

**Comment by rxaviers** [comment]
Yeap, me too. This line enables https://github.com/kborchers/react-globalize/pull/7/files#diff-406807581e0b49f296c2beb81859ba21R51, which renders the element (assuming it's a GlobalizeReact element) and get its output.

**Comment by kborchers** [comment]
Yeah, I know what it does I just don't like how it does it. Directly calling render() rather than letting React handle rendering "feels" wrong but not sure if there's a better way.

**Comment by rxaviers** [comment]
I have no idea how to make that hack look better. But, one way to certainly get rid of it is to, well, get rid of it :smiley:. Instead of allowing ReactGlobalize elements to be passed as options, we allow Globalize.<fn> outputs:

```diff
task count 1000 formatted - <FormatMessage ref="formattedTask" locale={this.state.locale} path="task" variables={{
   count: 1000,
-   formattedCount: <FormatNumber locale={this.state.locale} value={1000} />
+   formattedCount: Globalize(this.state.locale).formatNumber(1000) />
}} />
```

**Comment by kborchers** [comment]
Yeah, I think that's what I had before and I think I prefer that way for now.

**Comment by rxaviers** [comment]
I will change/revert and let you know.

**Comment by kborchers** [comment]
I would like to merge this. Can you revert that bit we talked about and ping me and I'll merge?

**Comment by rxaviers** [comment]
Reverted

**Comment by kborchers** [comment]
Thanks landed

**Pull request closed without merging**